presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/PartitionFilter.java

```diff
@@ -30,38 +27,35 @@
 public class PartitionFilter
 {
     private final HiveTableName hiveTableName;
-    private final List<String> parts;
+    private final TupleDomain<Column> effectivePredicate;
```

Comment:
`PartitionFilter` with `TupleDomain` is not Json serializable.
So far the only usages of this class to serialize/deserialize was in `RecordingHiveMetastore` where a string representation has been used now. 
In case we want to keep supporting this as a Json serializable object then we need to add in a custom ser/deserialization mechanism

Reply:
I would suggest you to move the `BlockJsonSerde` to the SPI, and register Serializer / Deserializer for Json encode used in the `RecordingHiveMetastore`. If you decide to proceed this way, please do the required moves as a separate commit.

Reply:
As moving `BlockJsonSerde` is a heavy-weight operation we will rely on the `toString` for now instead of trying to Json serialize this.